ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPoolTest.java

```diff
@@ -121,12 +123,33 @@ public void getAttestationsForBlock_shouldNotAddMoreAttestationsThanAllowedInBlo
     final Attestation attestation1 = addAttestationFromValidators(attestationData, 1, 2, 3, 4);
     final Attestation attestation2 = addAttestationFromValidators(attestationData, 2, 5);
     // Won't be included because of the 2 attestation limit.
-    addAttestationFromValidators(attestationData, 2, 6);
+    addAttestationFromValidators(attestationData, 2);
 
     assertThat(aggregatingPool.getAttestationsForBlock(SLOT))
         .containsExactly(attestation1, attestation2);
   }
 
+  @Test
+  public void onSlot_shouldPruneAttestationsMoreThanTwoEpochsBehindCurrentSlot() {
+    final AttestationData pruneAttestationData = randomAttestationDataToIncludeInBlock();
```

Comment:
nit: I was almost writing: 
> It would be nice to explicitly set the slot of the `pruneAttestationData` here. Otherwise, what if randomly it has a higher slot than the `preserveAttestationData`? AggregatingPool would clean both of them.

Then noticed the `randomAttestationDataToIncludeInBlock` method actually sets the slot to `SLOT - 1`. It's a bit hard to see especially when looking at git diffs.